front/src/modules/powerRestriction/helpers/powerRestrictionWarnings.ts

```diff
@@ -0,0 +1,96 @@
+ * returns the possible warnings based on the incompatibilty between them on each interval
+ * @returns A dictionnary of warnings ordered by power restriction codes
+ */
+export const getPowerRestrictionsWarnings = (
+  powerRestrictionRanges: RangedValue[],
+  pathElectrificationRanges: RangedValue[],
+  rollingStockModes: RollingStock['effort_curves']['modes']
+): PowerRestrictionWarnings => {
+  const powerRestrictionsByMode = getRollingStockPowerRestrictionsByMode(rollingStockModes);
+  const invalidCombinationWarnings: PowerRestrictionWarnings['invalidCombinationWarnings'] = [];
+  const modeNotSupportedWarnings: PowerRestrictionWarnings['modeNotSupportedWarnings'] = [];
+  const missingPowerRestrictionWarnings: PowerRestrictionWarnings['missingPowerRestrictionWarnings'] =
+    [];
+
+  powerRestrictionRanges.forEach((powerRestrictionRange) => {
+    // find path ranges crossed or included in the power restriction range
+    pathElectrificationRanges.forEach((electrificationRange) => {
+      // no intersection between the path range and the power restriction range
+      if (
+        electrificationRange.end <= powerRestrictionRange.begin ||
+        powerRestrictionRange.end <= electrificationRange.begin
+      )
+        return;
+
+      const { value: mode } = electrificationRange;
+      const invalidZoneBoundaries = getInvalidZoneBoundaries(
+        powerRestrictionRange,
+        electrificationRange
+      );
+      if (mode in powerRestrictionsByMode) {
+        const powerRestrictions = powerRestrictionsByMode[mode];
+        const isInvalid = !powerRestrictions.includes(powerRestrictionRange.value);
+        if (isInvalid) {
+          if (powerRestrictionRange.value === NO_POWER_RESTRICTION) {
+            missingPowerRestrictionWarnings.push({
+              ...invalidZoneBoundaries,
+            });
+          } else {
+            invalidCombinationWarnings.push({
+              ...invalidZoneBoundaries,
+              powerRestrictionCode: powerRestrictionRange.value,
+              electrification: mode,
+            });
+          }
+        }
+      } else if (powerRestrictionRange.value !== NO_POWER_RESTRICTION) {
+        // the mode is not supported by the rolling stock, so no power restriction should be defined
+        // (the rolling stock uses thermal propulsion)
+        modeNotSupportedWarnings.push({
+          ...invalidZoneBoundaries,
+          electrification: mode,
+        });
+      }
+    });
+  });
+  return {
+    invalidCombinationWarnings,
+    modeNotSupportedWarnings,
+    missingPowerRestrictionWarnings,
+  };
+};
+
+export const countWarnings = (warnings?: PowerRestrictionWarnings): number => {
+  if (!warnings) return 0;
+  const { invalidCombinationWarnings, modeNotSupportedWarnings, missingPowerRestrictionWarnings } =
+    warnings;
+  return (
+    invalidCombinationWarnings.length +
+    modeNotSupportedWarnings.length +
+    missingPowerRestrictionWarnings.length
+  );
+};
```

Comment:
```
export const countWarnings = (warnings: PowerRestrictionWarnings = { 
  invalidCombinationWarnings: [], 
  modeNotSupportedWarnings: [], 
  missingPowerRestrictionWarnings: [] 
}): number => {
  const { invalidCombinationWarnings, modeNotSupportedWarnings, missingPowerRestrictionWarnings } = warnings;
  return invalidCombinationWarnings.length + modeNotSupportedWarnings.length + missingPowerRestrictionWarnings.length;
};
```

this would be the same and it allow to skip the if check

Reply:
Done, it's clearer, thanks :pray:

Reply:
I prefer to keep the condition, I think it's more readable than the default value. But it's only a personal preference, is it ok for you ?

Reply:
Done